archive/import.go

```diff
@@ -161,48 +161,60 @@ func (d *importDriver) close() error {
 	return idxWriter.Close()
 }
 
-func (d *importDriver) Write(cid int, batch zbuf.Batch) error {
-	if cid != 0 {
-		panic("importDriver write to non-zero channel")
-	}
-	for i := 0; i < batch.Length(); i++ {
-		if err := d.writeOne(batch.Index(i)); err != nil {
+func (d *batchWriter) write(recs []*zng.Record) error {
+	expr.SortStable(recs, importCompareFn(d.ark))
+	for i := 0; i < len(recs); i++ {
+		if err := d.writeOne(recs[i]); err != nil {
 			return err
 		}
 	}
-	batch.Unref()
-	return nil
-}
-
-func (d *importDriver) ChannelEnd(cid int) error {
-	if cid != 0 {
-		panic("importDriver ChannelEnd to non-zero channel")
-	}
-	return d.close()
+	err := d.close()
+	return err
 }
 
-func (d *importDriver) Warn(warning string) error          { return nil }
-func (d *importDriver) Stats(stats api.ScannerStats) error { return nil }
-
-func importProc(ark *Archive) string {
-	if ark.DataSortDirection == zbuf.DirTimeForward {
-		return "sort ts"
+func importCompareFn(ark *Archive) expr.CompareFn {
+	return func(a, b *zng.Record) (cmp int) {
+		d := a.Ts() - b.Ts()
+		if d < 0 {
+			cmp = -1
+		}
+		if d > 0 {
+			cmp = 1
+		}
+		return cmp * ark.DataSortDirection.Int()
 	}
-	return "sort -r ts"
 }
 
 func Import(ctx context.Context, ark *Archive, zctx *resolver.Context, r zbuf.Reader) error {
-	proc, err := zql.ParseProc(importProc(ark))
-	if err != nil {
-		return err
+	maxBufSize := ImportBufSize
+	if maxBufSize < ark.LogSizeThreshold {
```

Comment:
I think it should be reversed: ark.LogSizeThreshold could be much greater (gigabytes) than the desired memory consumption for import:
```suggestion
	if maxBufSize > ark.LogSizeThreshold {
```

Reply:
@alfred-landrum I'm realizing this is causing test to fail that set the `ark.LogSizeThreshold` small. I'm inclined to think we should just get rid of this check all together for the time being.